### PR TITLE
[th2-1863] added toProtoEvent(EventID) and toProtoEvents(EventID) ove…

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**
…rloads into the Event class

### CODE REVIEW

**Review by jean-michel-perli-sonsin — CHANGES_REQUESTED** [review]
Hello, please see if you agree with the small changes suggested below:
README.md 
line 40 - connectionCloseTimeout - the timeout in milliseconds for completing all the close-related operations, use -1 for infinity, **the** default value is set to 10000.

line 42 - The `th2_readiness` probe is set to false and publishers are blocked after a lost connection to RabbitMQ. The `th2_readiness` probe is reverted to true if the connection will be recovered during specified attempts otherwise the `th2_liveness` probe will be set **to** false.

The rest of the text is correct.

Thanks,

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 1]
```diff
@@ -205,11 +207,27 @@ public Event messageID(MessageID attachedMessageID) {
...
 
+    /**
+     * @deprecated prefer to use full object instead of part of them, use the {@link #toProtoEvent(EventID)} method
+     */
+    @Deprecated
     public com.exactpro.th2.common.grpc.Event toProtoEvent(@Nullable String parentID) throws JsonProcessingException {
+        return toProtoEvent(toEventID(parentID));
+    }
+
+    public com.exactpro.th2.common.grpc.Event toProtoEvent(@Nullable EventID parentID) throws JsonProcessingException {
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/event/Event.java:230]: This new method will break the compatibility. If the original code calls `toProtoEvent(String)` with `null` parameter after adding that method the code won't compile. The compile can't choose between those two methods.

If it is okay and we are ready to change some places in code if we update the common-j version then ok, let's submit those changes

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 2]
```diff
@@ -205,11 +207,27 @@ public Event messageID(MessageID attachedMessageID) {
         return this;
     }
 
-    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(String parentID) throws JsonProcessingException {
+    /**
+     * @deprecated prefer to use full object instead of part of them, use the {@link #toListProto(EventID)} method
+     */
+    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(@Nullable String parentID) throws JsonProcessingException {
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/event/Event.java:213]: Missed `@Deprecated` annotation

File: src/main/java/com/exactpro/th2/common/event/Event.java [hunk 3]
```diff
@@ -247,10 +265,17 @@ public Instant getEndTimestamp() {
         return endTimestamp;
     }
 
+    /**
+     * @deprecated prefer to use full object instead of part of them, use the {@link #collectSubEvents(List, EventID)} method
+     */
     protected List<com.exactpro.th2.common.grpc.Event> collectSubEvents(List<com.exactpro.th2.common.grpc.Event> protoEvents, @Nullable String parentID) throws JsonProcessingException {
```
Comment (OptimumOpium) [inline src/main/java/com/exactpro/th2/common/event/Event.java:271]: Missed `@Deprecated` annotation

**Review by OptimumOpium — DISMISSED** [review]

**Review by OptimumOpium — APPROVED** [review]

**Pull request merged**